src/Authenticator/LoginHandler.php

```diff
@@ -614,32 +513,34 @@ public function getLogger()
         return $this->logger;
     }
 
-    /**
-     * @return StoreInterface
-     */
-    protected function getStore()
+    protected function applyRequirements(): void
     {
-        if (!$this->store) {
-            $this->store = Injector::inst()->create(StoreInterface::class, $this->getRequest());
-        }
+        // Run through requirements
+        Requirements::javascript('silverstripe/mfa: client/dist/js/injector.js');
+        Requirements::javascript('silverstripe/admin: client/dist/js/i18n.js');
+        Requirements::javascript('silverstripe/mfa: client/dist/js/bundle.js');
+        Requirements::add_i18n_javascript('silverstripe/mfa: client/lang');
+        Requirements::css('silverstripe/mfa: client/dist/styles/bundle.css');
 
-        return $this->store;
+        // Plugin module requirements
+        $registry = $this->getMethodRegistry();
+        foreach ($registry->getMethods() as $method) {
+            $method->applyRequirements();
+        }
     }
 
     /**
-     * @return MethodRegistry
+     * @return StoreInterface|null
      */
-    protected function getMethodRegistry()
+    protected function getStore(): ?StoreInterface
     {
-        return MethodRegistry::singleton();
-    }
+        if (!$this->store) {
+            $spec = Injector::inst()->getServiceSpec(StoreInterface::class);
+            $class = is_string($spec) ? $spec : $spec['class'];
+            $this->store = call_user_func([$class, 'load'], $this->getRequest());
```

Comment:
I think it'd be better to use an Injector factory class to create this instance

Reply:
Does this mean I'd have to hackily fetch the request from Injector? I'm getting flashbacks to https://github.com/silverstripe/silverstripe-versioned-admin/blob/d9e83412a309cb0f03cdd50a218f4dc45b470821/src/Controllers/HistoryControllerFactory.php#L26

Reply:
I think I'd end up having similar logic in the factory too? I guess I'm not quite sure what this would look like.

Reply:
I guess it would to look up the class definition yeah. I guess the line this comment is on is the bit I'm confused about

Reply:
Yeah there wasn't really a good way to get the concrete implementation registered for an interface from `Injector`. This is just looking up whatever that is so that we can call the static `load` method. This will ensure it is fetched back from session. Maybe this is some code-smell. I'll think about it a little more...

Reply:
How about `new $class($this->getRequest())`? I removed the load method a while ago because it's basically a constructor, it'd be nice if we can keep it away

Reply:
The constructor is used to create the store with a member. The `load` method is used to fetch one that's been persisted into whatever storage - in `SessionStore`s case it's coming from the session.

Reply:
Okaaaaaay if you insist